tower-load/src/instrument.rs

```diff
@@ -59,18 +56,23 @@ where
     }
 }
 
-impl<F, I, H> Future for InstrumentFuture<F, I, H>
+impl<F, I, H, T, E> Future for InstrumentFuture<F, I, H>
 where
-    F: Future,
-    I: Instrument<H, F::Item>,
+    F: Future<Output = Result<T, E>>,
+    I: Instrument<H, T>,
```

Comment:
Why have the bounds changed so dramatically here?

Reply:
I am trying to avoid using `TryFuture` etc so in this case this replicates it via adding additional generics. This shouldn't affect the end user since F should ALWAYS allow T, E to be infered.